src/pokemon.c

```diff
@@ -4771,329 +4771,337 @@ u32 GetBoxMonData(struct BoxPokemon *boxMon, s32 field, u8 *data)
             boxMon->isEgg = TRUE;
             substruct3->isEgg = TRUE;
         }
-    }
 
-    switch (field)
-    {
-    case MON_DATA_PERSONALITY:
-        retVal = boxMon->personality;
-        break;
-    case MON_DATA_OT_ID:
-        retVal = boxMon->otId;
-        break;
-    case MON_DATA_NICKNAME:
-    {
-        if (boxMon->isBadEgg)
+        switch (field)
         {
-            for (retVal = 0;
-                retVal < POKEMON_NAME_LENGTH && gText_BadEgg[retVal] != EOS;
-                data[retVal] = gText_BadEgg[retVal], retVal++) {}
+            case MON_DATA_SPECIES:
+                retVal = boxMon->isBadEgg ? SPECIES_EGG : substruct0->species;
+                break;
+            case MON_DATA_HELD_ITEM:
+                retVal = substruct0->heldItem;
+                break;
+            case MON_DATA_EXP:
+                retVal = substruct0->experience;
+                break;
+            case MON_DATA_PP_BONUSES:
+                retVal = substruct0->ppBonuses;
```

Comment:
The cases should be aligned with the opening parenthesis of the `switch` statement.